src/Nethermind/Nethermind.KeyStore/BlockAuthorPasswordProvider.cs

```diff
@@ -0,0 +1,49 @@
+//  Copyright (c) 2018 Demerzel Solutions Limited
+//  This file is part of the Nethermind library.
+// 
+//  The Nethermind library is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+// 
+//  The Nethermind library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+
+using System;
+using System.Security;
+using Nethermind.KeyStore.Config;
+
+namespace Nethermind.KeyStore
+{
+    public class BlockAuthorPasswordProvider : IPasswordProvider
+    {
+        private readonly IKeyStoreConfig _keyStoreConfig;
+        private readonly PasswordProviderHelper _passwordProviderHelper;
+        private readonly KeyStorePasswordProvider _keyStorePasswordProvider;
+
+        public BlockAuthorPasswordProvider(IKeyStoreConfig keyStoreConfig, PasswordProviderHelper passwordProviderHelper)
```

Comment:
I dislike injecting PasswordProviderHelper as dependency. I like that it is reusable code, but injecting a class here doesn't achieve anything. Neither making code more testable, neither marking any explicit dependency as its only utils class.